### PR TITLE
Add Affine 

### PR DESCRIPTION
**Pull request by jjchiw**
First of all, thank you for your contribution! 😄

Please note that this repo is mostly for popular apps with thousands of stars and tens of thousands of downloads. If you'd like to add a less popular app, you can always [create your own 3rd party repo](https://github.com/caprover/one-click-apps#build-your-own-one-click-app-repository) and add your app there.

HomePage
https://affine.pro/

Documentation
https://docs.affine.pro/docs/self-host-affine

### ☑️ Self Check before Merge

- [X] I have tested the template using the method described in README.md thoroughly
- [X] I have ensured that I put as much default values as possible (except passwords) to ensure minimum effort required for end users to get started.
- [X] I have ensured that I am not using the "latest" tag as this tag is dynamically changing and might break the one-click app. Use a fixed version.
- [X] I have made sure that instructions.start and instructions.end are clear and self-explanatory.
- [X] Icon is added as a png file to the logos directory.
- [X] I've executed the checks if necessary by running `npm ci && npm run validate_apps && npm run formatter` (If failling run the prettier: `npm run formatter-write`)
- [X] I will take responsibility addressing any issues that arises as a result of this PR (maintaining this app).

### CODE REVIEW

File: public/v4/apps/affine.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
+captainVersion: 4
+services:
+  #affine:
+  $$cap_appname:
+    image: ghcr.io/toeverything/affine-graphql:$$cap_affine_version
+    command:
+      ['sh', '-c', 'node ./scripts/self-host-predeploy && node ./dist/index.js']
+    ports:
+      - '3010:3010'
```
Comment (githubsaturn) [inline public/v4/apps/affine.yml:9]: Looks like 3010 is a simple HTTP port - you can use `containerHttpPort`

File: public/v4/apps/affine.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      - '5555:5555'
```
Comment (githubsaturn) [inline public/v4/apps/affine.yml:10]: What is this used for? Can it be removed?

File: public/v4/apps/affine.yml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      timeout: 5s
+      retries: 5
+    environment:
+      POSTGRES_DB: affine
+      POSTGRES_USER: affine
+      POSTGRES_PASSWORD: $$cap_db_affine_password
+      PGDATA: /var/lib/postgresql/data/pgdata
+caproverOneClickApp:
+  variables:
+    - defaultValue: stable
```
Comment (githubsaturn) [inline public/v4/apps/affine.yml:73]: You should use an unmutable tag like `stable-4b9e2ab`

`stable` today is not same as `stable` tomorrow

File: public/v4/apps/affine.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+      validRegex: /^([^\s^\/])+$/
+
+    - description: Default access domain of AFFiNE. Make sure this matches the URL in browser.	localhost
+      id: $$cap_affine_server_host
+      label: AFFINE_SERVER_HOST
+
+    - defaultValue: 3010
+      description: Default access port of AFFiNE. Make sure this matches the URL in browser.	3010
+      id: $$cap_affine_server_port
+      label: AFFINE_SERVER_PORT
```
Comment (githubsaturn) [inline public/v4/apps/affine.yml:86]: Perhaps no need to be configured. Like I said above, container HTTP port can take of this.

File: public/v4/apps/affine.yml [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+    - description: Use https prefix in url. - Make sure this matches the URL in browser.	false
```
Comment (githubsaturn) [inline public/v4/apps/affine.yml:88]: A bit of unclear description. Is it a URL or just http/https? or true / false?

File: public/v4/apps/affine.yml [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+      PGDATA: /var/lib/postgresql/data/pgdata
+caproverOneClickApp:
+  variables:
+    - defaultValue: stable
+      description: Check out their Docker page for the valid tags https://github.com/toeverything/AFFiNE/pkgs/container/affine-graphql
+      id: $$cap_affine_version
+      label: Affine version
+      validRegex: /^([^\s^\/])+$/
+
+    - description: Default access domain of AFFiNE. Make sure this matches the URL in browser.	localhost
```
Comment (githubsaturn) [inline public/v4/apps/affine.yml:79]: I am guess `localhost` was supposed to be the default value and it is just a typo that it's placed at the end of the description?

**Review by githubsaturn — COMMENTED** [review]
Exciting app!

File: public/v4/apps/affine.yml (hunk 1 above)
Comment (jjchiw) [reply public/v4/apps/affine.yml]: Removed and added
```
    caproverExtra:
      containerHttpPort: '3010'
      websocketSupport: true
```

File: public/v4/apps/affine.yml (hunk 2 above)
Comment (jjchiw) [reply public/v4/apps/affine.yml:9]: I'm not sure why that port is used :/
https://github.com/toeverything/AFFiNE/discussions/5975#discussioncomment-10897370
I'll leave it from the time being

File: public/v4/apps/affine.yml (hunk 3 above)
Comment (jjchiw) [reply public/v4/apps/affine.yml]: I set the default value to the lastet stable version `stable-4b9e2ab`

File: public/v4/apps/affine.yml (hunk 4 above)
Comment (jjchiw) [reply public/v4/apps/affine.yml]: Removed

File: public/v4/apps/affine.yml (hunk 5 above)
Comment (jjchiw) [reply public/v4/apps/affine.yml]: Change to this
```
    - defaultValue: true
      description: Use https prefix in url. Make sure this matches the URL in browser.	(false/true)
      id: $$cap_affine_server_https
      label: AFFINE_SERVER_HTTP
```

File: public/v4/apps/affine.yml (hunk 6 above)
Comment (jjchiw) [reply public/v4/apps/affine.yml]: It was the default value. I changed to `(eg. yourdomain.com)`

**Comment by githubsaturn** [comment]
Thanks!

**Pull request merged**